### PR TITLE
fix(cloudweb): display configuration usage in disk usage

### PR DESCRIPTION
**Pull request by cbourgois**
### Description of the Change

Display configuration usage for CloudWeb in disk usage progressbar.

### CODE REVIEW

File: client/app/hosting/general-informations/GENERAL_INFORMATIONS.html [hunk 1]
```diff
@@ -384,14 +384,59 @@ <h3 class="oui-tile__title" data-i18n-static="hosting_dashboard_configuration"><
...
+                            data-ng-class="{
+                                'oui-progress_success': hosting.quotaPercent + hosting.configurationPercent < 51,
+                                'oui-progress_warning': hosting.quotaPercent + hosting.configurationPercent > 50 && hosting.quotaPercent + hosting.configurationPercent < 81,
+                                'oui-progress_error': hosting.quotaPercent + hosting.configurationPercent > 80
+                            }">
+                            <div
+                                class="oui-progress__bar oui-progress__bar_info"
+                                role="progressbar"
+                                data-ng-style="{'width': hosting.configurationPercent + '%'}"
+                                aria-valuenow="{{hosting.configurationPercent}}"
```
Comment (FredericEspiau) [inline client/app/hosting/general-informations/GENERAL_INFORMATIONS.html]: `::`

File: client/app/hosting/general-informations/GENERAL_INFORMATIONS.html [hunk 2]
```diff
@@ -384,14 +384,59 @@ <h3 class="oui-tile__title" data-i18n-static="hosting_dashboard_configuration"><
...
                     <strong data-ng-bind-html="i18n.hosting_dashboard_disk_usage"></strong>
-                    <div class="oui-progress mt-2"
-                         data-ng-class="{
-                            'oui-progress_success': hosting.quotaPercent < 51,
+
+                    <div data-ng-if="hosting.isCloudWeb">
+                        <div
+                            class="oui-progress mt-2"
+                            data-ng-class="{
+                                'oui-progress_success': hosting.quotaPercent + hosting.configurationPercent < 51,
```
Comment (FredericEspiau) [inline client/app/hosting/general-informations/GENERAL_INFORMATIONS.html]: Percentage

File: client/app/hosting/general-informations/GENERAL_INFORMATIONS.html [hunk 3]
```diff
@@ -384,14 +384,59 @@ <h3 class="oui-tile__title" data-i18n-static="hosting_dashboard_configuration"><
...
+                                data-ng-style="{'width': hosting.configurationPercent + '%'}"
+                                aria-valuenow="{{hosting.configurationPercent}}"
+                                aria-valuemin="0"
+                                aria-valuemax="100">
+                            </div>
+                            <div
+                                class="oui-progress__bar"
+                                role="progressbar"
+                                data-ng-class="{
+                                    'oui-progress__bar_success': hosting.quotaPercent + hosting.configurationPercent < 51,
```
Comment (FredericEspiau) [inline client/app/hosting/general-informations/GENERAL_INFORMATIONS.html]: Percentage

File: client/app/hosting/general-informations/GENERAL_INFORMATIONS.html [hunk 4]
```diff
@@ -384,14 +384,59 @@ <h3 class="oui-tile__title" data-i18n-static="hosting_dashboard_configuration"><
...
+                                aria-valuemax="100">
+                                <span
+                                    class="oui-progress__label"
+                                    data-ng-bind="convertBytesSize(hosting.quotaUsed.value, hosting.quotaUsed.unit, 2) + ' / ' + convertBytesSize(hosting.quotaSize.value, hosting.quotaSize.unit)"></span>
+                            </div>
+                        </div>
+
+                        <p>
+
+                            <span class="oui-icon oui-icon_small oui-icon-shape-dot oui-icon_info-medium" aria-hidden="true"></span>
```
Comment (FredericEspiau) [inline client/app/hosting/general-informations/GENERAL_INFORMATIONS.html]: Link between the legend and the data for screen readers

File: client/app/css/less/ovh-ui-kit/icon.less [hunk 5]
```diff
@@ -4,4 +4,8 @@
     &.oui-icon_small {
         font-size: 1.4rem;
     }
+
+    &.oui-icon_info-medium {
```
Comment (antleblanc) [inline client/app/css/less/ovh-ui-kit/icon.less]: What's about having this directly on ovh-ux/ovh-ui-kit instead?

File: client/app/hosting/hosting.service.js [hunk 6]
```diff
@@ -169,6 +171,23 @@
                     clearCache: forceRefresh
                 }).then((hosting) => {
                     hosting.isCloudWeb = _(hosting.offer).includes("CLOUD");
+                    if (hosting.isCloudWeb) {
+                        hosting.configurationQuota = this.HOSTING.cloudWeb.configurationQuota;
+
+                        const configurationOctet = this.converterService.convertToOctet(hosting.configurationQuota.value, hosting.configurationQuota.unit);
+                        const quotaSizeOctet = this.converterService.convertToOctet(hosting.quotaSize.value, hosting.quotaSize.unit);
+
+                        hosting.configurationPercent = configurationOctet / quotaSizeOctet * 100;
```
Comment (FredericEspiau) [inline client/app/hosting/hosting.service.js]: Sure `quotaSizeOctet` can't be worth 0 ?

File: client/app/hosting/general-informations/GENERAL_INFORMATIONS.html [hunk 7]
```diff
@@ -384,14 +384,59 @@ <h3 class="oui-tile__title" data-i18n-static="hosting_dashboard_configuration"><
...
+                            <span data-ng-bind="tr('hosting_dashboard_disk_usage_configuration', [ convertBytesSize(hosting.configurationQuota.value, hosting.configurationQuota.unit, 2) ])"></span>
+
+                        </p>
+                    </div>
+
+                    <div
+                        class="oui-progress mt-2"
+                        data-ng-if="!hosting.isCloudWeb"
+                        data-ng-class="{
+                        'oui-progress_success': hosting.quotaPercent < 51,
```
Comment (antleblanc) [inline client/app/hosting/general-informations/GENERAL_INFORMATIONS.html]: indentation

Comment (FredericEspiau) [inline client/app/hosting/general-informations/GENERAL_INFORMATIONS.html]: Indent

File: client/app/css/less/ovh-ui-kit/icon.less (hunk 5 above)
Comment (cbourgois) [reply client/app/css/less/ovh-ui-kit/icon.less]: We have to discuss to have css helpers for colors (`background-color` / `color`).

File: client/app/css/less/ovh-ui-kit/icon.less [hunk 8]
```diff
@@ -4,4 +4,18 @@
     &.oui-icon_small {
         font-size: 1.4rem;
     }
+
+    &.oui-icon_progress-info-foreground-color {
```
Comment (jleveugle) [inline client/app/css/less/ovh-ui-kit/icon.less]: Why do you put this code here? It should be managed by ui-kit no?

File: client/app/hosting/general-informations/GENERAL_INFORMATIONS.html [hunk 9]
```diff
@@ -384,28 +384,63 @@ <h3 class="oui-tile__title" data-i18n-static="hosting_dashboard_configuration"><
...
+                                class="oui-progress__label"
+                                data-ng-bind="convertBytesSize(hosting.quotaUsed.value, hosting.quotaUsed.unit, 2) + ' / ' + convertBytesSize(hosting.quotaSize.value, hosting.quotaSize.unit)"
+                                data-ng-if="!hosting.isCloudWeb"></span>
                         </div>
                     </div>
+
+                    <p id="diskUsageConfiguration"
+                       data-ng-if="hosting.isCloudWeb">
+                        <span class="oui-icon oui-icon_small oui-icon-shape-dot oui-icon_progress-info-foreground-color" aria-hidden="true"></span>
+                        <span data-ng-bind="tr('hosting_dashboard_disk_usage_configuration', [ convertBytesSize(hosting.configurationQuota.value, hosting.configurationQuota.unit, 2) ])"></span>
```
Comment (jleveugle) [inline client/app/hosting/general-informations/GENERAL_INFORMATIONS.html]: `span` + `br` + `span` === `div` + `div`

File: client/app/hosting/hosting.service.js [hunk 10]
```diff
@@ -9,12 +9,14 @@
...
         .service("Hosting", class Hosting {
 
-            constructor ($q, $http, $rootScope, $stateParams, constants, OvhHttp, Poll, Products) {
+            constructor ($q, $http, $rootScope, $stateParams, constants, ConverterService, HOSTING, OvhHttp, Poll, Products) {
                 this.$q = $q;
                 this.$http = $http;
                 this.$rootScope = $rootScope;
                 this.$stateParams = $stateParams;
                 this.constants = constants;
+                this.converterService = ConverterService;
```
Comment (jleveugle) [inline client/app/hosting/hosting.service.js]: `converterService` vs `ConverterService`

File: client/app/css/less/ovh-ui-kit/icon.less (hunk 8 above)
Comment (cbourgois) [reply client/app/css/less/ovh-ui-kit/icon.less]: We have to discuss to have css colors helpers (background-color / color) in ui-kit.

File: client/app/hosting/hosting.service.js [hunk 11]
```diff
@@ -170,6 +172,17 @@
                 }).then((hosting) => {
                     hosting.isCloudWeb = _(hosting.offer).includes("CLOUD");
 
+                    if (hosting.isCloudWeb) {
+                        hosting.configurationQuota = this.HOSTING.cloudWeb.configurationQuota;
+                        hosting.totalQuota = _.clone(this.HOSTING.cloudWeb.configurationQuota);
+
+                        const configurationOctet = this.ConverterService.convertToOctet(hosting.configurationQuota.value, hosting.configurationQuota.unit);
+                        const quotaSizeOctet = this.ConverterService.convertToOctet(hosting.quotaSize.value, hosting.quotaSize.unit);
```
Comment (euhmeuh) [inline client/app/hosting/hosting.service.js]: **octet** (plural octets)
A group of eight musicians performing together. 

:notes:

Comment (cbourgois) [reply client/app/hosting/hosting.service.js]: this.ConverterService.convertToOctet => octet

**Pull request merged**